ServerCore/Helpers/TeamHelper.cs

```diff
@@ -35,6 +36,16 @@ public static async Task CreateTeamAsync(PuzzleServerContext context, Team team,
                         throw new Exception("You are already on a team. Leave that team before creating a new one");
                     }
 
+                    if (team.IsRemoteTeam && await context.Teams.Where((t) => t.Event == ev && t.IsRemoteTeam).CountAsync() >= ev.MaxNumberOfRemoteTeams)
+                    {
+                        throw new Exception("Remote team registration is full. No further remote teams may be created at the present time.");
+                    }
+
+                    if (!team.IsRemoteTeam && await context.Teams.Where((t) => t.Event == ev && !t.IsRemoteTeam).CountAsync() >= ev.MaxNumberOfLocalTeams)
+                    {
+                        throw new Exception("Local team registration is full. No further local teams may be created at the present time.");
+                    }
+
                     if (await context.Teams.Where((t) => t.Event == ev).CountAsync() >= ev.MaxNumberOfTeams)

```

Comment:
Unfortunate that all three of these max numbers exist, and all three need to be set, and if you forget one then you get weird results. Can we just have two properties plus math to get the third, perhaps by deleting MaxNumberOfLocalTeams and computing it as MaxNumberOfTeams - MaxNumberOfRemoteTeams?

Reply:
(they'll break because the local and remote maxes will be zero, effectively locking registration)

Reply:
right you are, sorry, I caught that in the other check but not in this one, I'll tweak things.